### ISSUE TITLE
Sentence Type Semantic Effects: Learn illocutionary force

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#104]
## Overview

Sentence type (question/command/statement) carries semantic meaning about **what the speaker is doing** (illocutionary force), not just structural variation.

## Current State

**In AST**: ❌ NOT CAPTURED

Issue #87 tracks detecting sentence type, but we also need to learn its semantic effect.

### Illocutionary Force
- **Statement (aserto)**: Asserts truth, provides information
- **Question (demando)**: Seeks information, expresses uncertainty
- **Command (ordono)**: Expresses desire for action, directive

### Examples
- "Li venas" (He is coming) - assertion of fact
- "Ĉu li venas?" (Is he coming?) - uncertainty, seeks confirmation
- "Venu\!" (Come\!) - desire for action

### For Q&A Systems
- Questions in corpus might be rhetorical or seeking different info
- Statements are facts to retrieve
- Commands express intent, not facts

### Same Content, Different Force
- "Frodo havas la Ringon" - fact
- "Ĉu Frodo havas la Ringon?" - question about this fact
- Different embeddings appropriate\!

### 1. Detection (Deterministic) - Issue #87
```python
def detect_sentence_type(ast, original_text):
    if original_text.endswith('?'):
        return 'demando'
    if original_text.endswith('\!'):
        # Check if imperative verb
        if ast['verbo'] and ast['verbo'].get('modo') == 'imperativo':
            return 'ordono'
        return 'emfazo'  # Exclamatory statement
    if any(w.get('radiko') == 'ĉu' for w in ast['aliaj']):
        return 'demando'
    if ast['verbo'] and ast['verbo'].get('modo') == 'imperativo':
        return 'ordono'
    return 'aserto'
```

### 2. Semantic Effect (Learned)
```python
class SentenceTypeTransformer(nn.Module):
    def __init__(self, dim):
        self.type_transforms = nn.ModuleDict({
            'aserto': nn.Identity(),  # Baseline
            'demando': nn.Linear(dim, dim),  # Question
            'ordono': nn.Linear(dim, dim),   # Command
        })
    
    def forward(self, embedding, sentence_type):
        return self.type_transforms[sentence_type](embedding)
```

### 3. Training Pairs
```python
# Statement vs question about same content
("Li venas", "Ĉu li venas?", similarity=0.5)

# Questions should cluster
("Ĉu li venas?", "Kiu venas?", similarity=0.7)

# Commands should cluster
("Venu\!", "Iru\!", similarity=0.6)

# Statement vs command
("Li venas", "Venu\!", similarity=0.3)
```

### 4. Evaluation Metrics
| Metric | Target |
|--------|--------|
| Sentence type classification | >95% |
| Question vs statement discrimination | >85% |
| Appropriate retrieval by type | Match question to answer |

## Dependencies

- #87 Sentence type detection (structural prerequisite)

## Related Issues

- #101 Meta: Grammatical Features with Semantic Content (parent)
- #87 Sentence type detection (detection in parser)
- #103 Mood semantics (commands relate to imperative)
